internal/model/labels.go

```diff
@@ -59,6 +61,53 @@ func (l Labels) Clone() Labels {
 	return cp
 }
 
+// Equal returns true when the two maps of labels are equal.
+func (l Labels) Equal(labels Labels) bool {
+	for key, localV := range l {
+		v, ok := labels[key]
+		if !ok {
+			return false
+		}
+		if v.Value != localV.Value {
+			return false
+		}
+		if len(v.Values) != len(localV.Values) {
+			return false
+		}
+		for _, s := range [][]string{v.Values, localV.Values} {
+			if !sort.StringsAreSorted(s) {
+				sort.Strings(s)
+			}
+		}
+		for i, v := range v.Values {
+			if v != localV.Values[i] {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+// Flatten returns a string interpretation of the labels.
+func (l Labels) Flatten() string {
```

Comment:
Until we find another need for flattening labels, can we keep this in the txmetrics code?
Also, it would be better not to build a string only to write it to a hash, as it will create garbage: instead pass in an io.Writer.